features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdConfigReloadIT.java

```diff
@@ -28,91 +28,115 @@
+		"classpath:/org/opennms/netmgt/trapd/applicationContext-trapDaemonTest.xml"
 })
 @JUnitConfigurationEnvironment
-public class TrapdConfigReloadIT extends CamelBlueprintTest {
+@JUnitTemporaryDatabase
+public class TrapdConfigReloadIT {
 
-	private AtomicInteger m_port = new AtomicInteger(1162);
+	@Autowired
+	private TrapdConfigFactory trapdConfig;
 
 	@Autowired
-	private DistPollerDao distPollerDao;
+	private Trapd trapd;
 
-	@Override
-	protected String setConfigAdminInitialConfiguration(Properties props) {
-		getAvailablePort(m_port, 2162);
-		props.put("trapd.listen.port", String.valueOf(m_port.get()));
-		return "org.opennms.netmgt.trapd";
-	}
+	@Autowired
+	private TrapListener trapListener;
 
-	@Override
-	protected void addServicesOnStartup(Map<String, KeyValueHolder<Object, Dictionary>> services) {
-		final RestClient client;
-		try {
-			client = mock(RestClient.class);
-			when(client.getSnmpV3Users()).thenReturn("<?xml version='1.0'?>"
-				+ "<trapd-configuration xmlns='http://xmlns.opennms.org/xsd/config/trapd' snmp-trap-address='127.0.0.1' snmp-trap-port='10500' new-suspect-on-trap='true'>"
-				+ "     <snmpv3-user security-name='opennms' security-level='0' auth-protocol='MD5' auth-passphrase='0p3nNMSv3' privacy-protocol='DES' privacy-passphrase='0p3nNMSv3' />"
-				+ "</trapd-configuration>");
-		} catch (Exception e) {
-			throw Throwables.propagate(e);
-		}
-		// add mocked services to osgi mocked container (Felix Connect)
-		services.put(RestClient.class.getName(), asService(client, null, null));
-		services.put(MessageConsumerManager.class.getName(), asService(new MockMessageConsumerManager(), null, null));
-		services.put(MessageDispatcherFactory.class.getName(), asService(new MockMessageDispatcherFactory<>(), null, null));
-		services.put(DistPollerDao.class.getName(), asService(distPollerDao, null, null));
+	@Autowired
+	private MockEventIpcManager events;
+
+	private final static InetAddress LOCALHOST = InetAddressUtils.addr("127.0.0.1");
+
+	@Before
+	public void setUp() {
+		this.events.setSynchronous(true);
+		this.events.getEventAnticipator().reset();
+
+		this.trapd.onStart();
 	}
 
-	// The location of our Blueprint XML files to be used for testing
-	@Override
-	protected String getBlueprintDescriptor() {
-		return "OSGI-INF/blueprint/blueprint-trapd-listener.xml";
+	@After
+	public void tearDown() {
+		this.trapd.onStop();
+
+		this.events.getEventAnticipator().verifyAnticipated();
 	}
 
 	@Test
-	public void verifyReload() throws Exception {
-		// Check that it has not yet been refreshed
-		TrapdConfig trapdConfig = getOsgiService(TrapdConfig.class);
-		Assert.assertEquals(true, trapdConfig.getSnmpV3Users().isEmpty());
+	public void testSnmpV3UserUpdate() throws Exception {
+		final var user = this.trapdConfig.getConfig().getSnmpv3User(0);
+
+		this.events.getEventAnticipator().anticipateEvent(new EventBuilder("uei.opennms.org/default/trap", "trapd")
+																  .setInterface(LOCALHOST).getEvent());
```

Comment:
Should this be formatted this way?

Reply:
This looks wired because this files is so old, it still uses tabs instead of spaces.